### PR TITLE
chore: Update crates

### PR DESCRIPTION
**Pull request by luke-biel**
Major changes are:

json_schema::ValidationError replaces CompilationError. As it borrows now the original schema, I had to stringify it.
There was bigger change in tonic::Interceptors, I had to change fn signatures.
I've changed `OB`, now it uses common `ER` and `SR` pools.

Additionally updated rust toolchain while i was at it, to ~~2021-07-13~~ 2021-06-30.

### CODE REVIEW

**Review by esavier — DISMISSED** [review]

File: crates/utils/crates/tracing/src/grpc.rs [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
+
 /// Method used with gRPC client to inject Span ID to gRPC metadata.
 /// Used mostly in `rpc` crate in `connect()`.
 /// # Example:
 /// ```ignore
 /// pub async fn connect(addr: String) -> Result<MyServiceClient<tonic::transport::Channel>, tonic::transport::Error> {
 ///     let conn = tonic::transport::Endpoint::new(addr)?.connect().await?;
 ///
-///     Ok(MyServiceClient::with_interceptor(conn, tracing_utils::grpc::interceptor()))
+///     Ok(MyServiceClient::with_interceptor(conn, tracing_utils::grpc::interceptor)))
```
[the remark this answers is not in the document]
Comment (luke-biel) [reply crates/utils/crates/tracing/src/grpc.rs]: Fixed

File: crates/rpc/src/schema_registry/types.rs [hunk 2]
```diff
@@ -1,15 +1,15 @@
-use std::convert::TryFrom;
-
-use async_graphql::Enum;
-use derive_more::Display;
-use serde::{Deserialize, Serialize};
+#![allow(clippy::nonstandard_macro_braces)] // There's bug in clippy: https://github.com/rust-lang/rust-clippy/issues/7434
```
Comment (luke-biel) [inline crates/rpc/src/schema_registry/types.rs:1]: There's a bug in clippy that has forced me to include this line.

**Review by frondeus — APPROVED** [review]

File: crates/rpc/src/schema_registry/types.rs [hunk 3]
```diff
@@ -1,15 +1,13 @@
-use std::convert::TryFrom;
-
-use async_graphql::Enum;
-use derive_more::Display;
-use serde::{Deserialize, Serialize};
-
 use super::logic_operator;
 use super::schema_type;
 use super::search_for;
+use async_graphql::Enum;
```
Comment (smores56) [inline crates/rpc/src/schema_registry/types.rs:4]: The original import organization in this file of STD, then external, then internal is preferred.

**Review by smores56 — DISMISSED** [review]

File: crates/rpc/src/schema_registry/types.rs (hunk 3 above)
Comment (luke-biel) [reply crates/rpc/src/schema_registry/types.rs:4]: Do we have a tool for that? I ran rustfmt.

File: crates/rpc/src/edge_registry.rs [hunk 4]
```diff
@@ -1,31 +1,38 @@
...
 
 pub struct EdgeRegistryConnectionManager {
     pub address: String,
 }
 
-pub async fn connect(addr: String) -> Result<EdgeRegistryClient<Channel>, ClientError> {
+pub async fn connect(
+    addr: String,
+) -> Result<
+    EdgeRegistryClient<InterceptedService<Channel, &'static dyn InterceptorType>>,
```
Comment (Kononnable) [inline crates/rpc/src/edge_registry.rs:20]:
```suggestion
    EdgeRegistryConn,
```

File: crates/rpc/src/edge_registry.rs [hunk 5]
```diff
@@ -1,31 +1,38 @@
...
     connect_inner(addr)
         .await
         .map_err(|err| ClientError::ConnectionError { source: err })
 }
 
 async fn connect_inner(
     addr: String,
-) -> Result<EdgeRegistryClient<Channel>, tonic::transport::Error> {
-    let conn = tonic::transport::Endpoint::new(addr)?.connect().await?;
+) -> Result<EdgeRegistryClient<InterceptedService<Channel, &'static dyn InterceptorType>>, Error> {
```
Comment (Kononnable) [inline crates/rpc/src/edge_registry.rs:30]:
```suggestion
) -> Result<EdgeRegistryConn, Error> {
```

File: crates/rpc/src/materializer_ondemand.rs [hunk 6]
```diff
@@ -1,31 +1,41 @@
...
     connect_inner(addr)
         .await
         .map_err(|err| ClientError::ConnectionError { source: err })
 }
 
 async fn connect_inner(
     addr: String,
-) -> Result<OnDemandMaterializerClient<Channel>, tonic::transport::Error> {
+) -> Result<
+    OnDemandMaterializerClient<InterceptedService<Channel, &'static dyn InterceptorType>>,
```
Comment (Kononnable) [inline crates/rpc/src/materializer_ondemand.rs:31]:
```suggestion
    OnDemandMaterializerConn,
```

**Review by Kononnable — COMMENTED** [review]
Those new type aliases(e.g. `EdgeRegistryConn`) exist in some related files, not all. It may be worth standardizing it(all or none) - apart from the existence of this type alias each of those files looks almost the same.

File: crates/rpc/src/schema_registry/types.rs (hunk 3 above)
Comment (luke-biel) [reply crates/rpc/src/schema_registry/types.rs]: I have a tool in CLion that does this to me, but afaik, there's no command line utility that could enforce this via GH Actions, even though I agree with you that we may want to keep this order.

**Review by frondeus — APPROVED** [review]

**Review by esavier — APPROVED** [review]

**Pull request merged**